jsonargparse_tests/test_core.py

```diff
@@ -60,6 +79,12 @@ def test_parse_args_unrecognized_arguments(parser):
     assert "Unrecognized arguments:" in err
 
 
+def test_parse_args_from_sys_argv(parser, auto_recover_sys_argv):
+    parser.add_argument("--op", type=str)
+    sys.argv = ["", "--op", "hello"]
+    assert parser.parse_args() == Namespace(op="hello")
```

Comment:
Awesome!

Just one comment about the fixture. I think it is unnecessary. The same can be done like:

```suggestion
    with patch("sys.argv", ["", "--op", "hello"]):
        assert parser.parse_args() == Namespace(op="hello")
```

Reply:
Sure, thank you very much! a lot learned :)